README.md

```diff
@@ -1,2 +1,72 @@
 # napari-plugin-devtools
-A repo with tools and services for napari plugin developers
+
+[![License](https://img.shields.io/pypi/l/napari-plugin-devtools.svg?color=green)](https://github.com/napari/napari-plugin-devtools/raw/main/LICENSE)
+[![PyPI](https://img.shields.io/pypi/v/napari-plugin-devtools.svg?color=green)](https://pypi.org/project/napari-plugin-devtools)
+[![Python Version](https://img.shields.io/pypi/pyversions/napari-demo.svg?color=green)](https://python.org)
+[![tests](https://github.com/napari/napari-plugin-devtools/workflows/CI/badge.svg)](https://github.com/napari/napari-plugin-devtools/actions)
+
+A repo with tools and services for [napari plugin](https://napari.org/docs/dev/plugins/index.html) developers
+
+## Validation tool
+The validation tools provides more transparency and confidence for 
+plugin developers through automated tests, these checks verify that 
+the plugin is available for users of napari to install, and would 
+register entry points with napari. 
+
+There are two parts to the validation 
+tool, one part being CLI and another part being pytest fixture, 
+
+### CLI Usage
+CLI can be used by CI/CD pipelines to introduce quick verifications 
+of plugin setups without more specific input required, it serves as
+a "sanity check". (It is can accessible from python in validation.py)
+
+The tool starts as `npd <cmd>`, there are 2 cmds currently:
+
+`npd --validate-packages` should be used after packages are built
+for pypi releases, (example: this repo is built by pep517 standard
+`pip install pep517 && python -m pep517.build .`) and the validator
+checks that all packages built under dist are correctly tagged with
+classifier "Framework :: napari". This is required for most plugins
+unless you wish your plugin remain hidden to typical end users.
+
+`npd --validate-functions` verify that after the plugin is installed 
+to current python environment, typically done through `pip install -e .`, 
+there are at least one analysis function registered by the plugin, and 
+there are no conflicts in name or registration error, it also output 
+all registered function signaturesin a json format that can be used 
+for further inspection, for example:
+```
+    [{
```

Comment:
this is great!  might be nice to use `inspect.signature` type output instead of `getfullargspec` style.  somehing like:
```json
{
    "arguments": {
        "layerA": "napari.types.ImageData"
    },
    "return_annotation": "napari.types.ImageData"
}
```

Reply:
Agreed, this could be nice. I don't have a strong preference though. I've merged the readme as we'd need to change this both in code and readme which would be a follow-up PR.